File: Prototype/Python/Principal.py
from Circulo import *
from Cuadrado import *
class Principal:
    circulo = Circulo("Mi circulo", 10, 10)
    cuadrado = Cuadrado("Mi cuadrado", 15, 15)

    print("Informacion de las figuras originales")
    
    print(circulo.getNombre())
    circulo.getPosicion()

    print(cuadrado.getNombre())
    cuadrado.getPosicion()

    print("Informacion de las figuras clonadas")

    clonCirculo = circulo.clonar()
    clonCirculo.setNombre("Clon del circulo")
    clonCirculo.mover(11, 11)
    print(clonCirculo.getNombre())
    clonCirculo.getPosicion()
 

    clonCuadrado = cuadrado.clonar()
    clonCuadrado.setNombre("Clon del cuadrado")
    clonCuadrado.mover(16, 16)
    print(clonCuadrado.getNombre())
    clonCuadrado.getPosicion()

File: Prototype/Java/Principal.java


public class Principal {

    public static void main(String [] args)
    {
        Circulo circulo = new Circulo("Mi circulo", 10, 10);
        Cuadrado cuadrado = new Cuadrado("Mi rectanuglo", 15, 15);

        System.out.println("Informacion del circulo");
        System.out.println(circulo.getNombre());
        circulo.getPosicion();


        System.out.println("Informacion del cuadrado");
        System.out.println(cuadrado.getNombre());
        cuadrado.getPosicion();

        // clonando 
        Figura figura;
        figura = circulo.clonar();
        figura.setNombre("Figura clon del circulo");

        System.out.println(figura.getNombre());
        figura.mover(10,-10);
        figura.getPosicion();;

    }
    
}
